CompilerSource/gcc_interface/gcc_backend.cpp

```diff
@@ -162,8 +139,8 @@ const char* establish_bearings(const char *compiler)
   /* Get a list of all macros defined by our compiler.
   ** These will help us through parsing available libraries.
   ***********************************************************/
-  if ((cmd = compey.get("defines")) == "")
-    return (sprintf(errbuf,"Compiler descriptor file `%s` does not specify 'defines' executable.\n", compfq.c_str()), errbuf);
+  if ((cmd = compiler_map_get("DEFINES", compilerInfo.parser_vars)) == "")
```

Comment:
This could be compile-time checked if you declared a variable for it within compilerInfo instead of trying to be general with a map. I guess you haven't made it worse, really... not not much better. Also, I believe e-yaml already does a `toLower` transformation on this stuff, so your uppercase transform was unnecessary, iirc.